verticapy/learn/memmodel.py

```diff
@@ -0,0 +1,1036 @@
+                  intercept: float, 
+                  method: str = "LinearRegression",):
+    """
+    ---------------------------------------------------------------------------
+    Returns the SQL code needed to deploy a linear model using its attributes.
+
+    Parameters
+    ----------
+    X: list
+        The name or values of the input predictors.
+    coefficients: list
+        List of the model's coefficients.
+    intercept: float
+        The intercept or constant value.
+    method: str, optional
+        The model category, one of the following: 'LinearRegression', 'LinearSVR', 
+        'LogisticRegression', or 'LinearSVC'.
+
+    Returns
+    -------
+    str
+        SQL code
+    """
+    check_types([("X", X, [list],), 
+                 ("coefficients", coefficients, [list],),
+                 ("intercept", intercept, [float, int,],),
+                 ("method", method, ["LinearRegression", "LinearSVR", "LogisticRegression", "LinearSVC"],),])
+    assert len(X) == len(coefficients), ParameterError("The length of parameter 'X' must be equal to the number of coefficients.")
+    sql = [str(intercept)] + [f"{coefficients[idx]} * {(X[idx])}" for idx in range(len(coefficients))]
+    sql = " + ".join(sql)
+    if method in ("LogisticRegression",):
+        return f"1 / (1 + EXP(- ({sql})))"
+    elif method in ("LinearSVC",):
+        return f"1 - 1 / (1 + EXP({sql}))"
+    return sql
+
+# ---#
+def predict_from_bisecting_kmeans(X: Union[list, np.array], 
+                                  clusters: Union[list, np.array],
+                                  left_child: Union[list, np.array],
+                                  right_child: Union[list, np.array],
+                                  p: int = 2,):
+    """
+    ---------------------------------------------------------------------------
+    Predicts using a bisecting k-means model and the input attributes.
+
+    Parameters
+    ----------
+    X: list / numpy.array
+        The data on which to make the prediction.
+    clusters: list / numpy.array
+        List of the model's cluster centers.
+    left_child: list / numpy.array
+        List of the model's left children IDs. ID i corresponds to the left 
+        child ID of node i.
+    right_child: list / numpy.array
+        List of the model's right children IDs. ID i corresponds to the right 
+        child ID of node i.
+    p: int, optional
+        The p corresponding to the one of the p-distances.
+
+    Returns
+    -------
+    numpy.array
+        Predicted values
+    """
+    check_types([("X", X, [list,],), 
+                 ("clusters", clusters, [list,],),
+                 ("left_child", left_child, [list,],),
+                 ("right_child", right_child, [list,],),
+                 ("p", p, [int,],),])
+    centroids = np.array(clusters)
+    def predict_tree(right_child, left_child, row, node_id, centroids):
+        if left_child[node_id] == right_child[node_id] == None:
```

Comment:
This condition is ambiguous. What you have written is equivalent to `left_child[node_id] == right_child[node_id] or right_child[node_id] == None`. Did you want it to be `left_child[node_id] == right_child[node_id] or right_child[node_id] == None or left_child[node_id] == None`?

It would be always nice to avoid ambiguous coding.

Reply:
In Python it is possible to make multiple conditions.
left_child[node_id] == right_child[node_id] == None is equivalent to
left_child[node_id] == right_child[node_id] and right_child[node_id] == None